packages/explat-client/src/internal/test/requests.ts

```diff
@@ -0,0 +1,233 @@
+			Requests.fetchExperimentAssignment( mockedConfig, validExperimentAssignment.experimentName )
+		).rejects.toThrow( 'Invalid FetchExperimentAssignmentResponse' );
+	} );
+
+	it( 'should throw for multiple experiments in the response', async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( null, 1 ) );
+		spiedMonotonicNow.mockImplementationOnce( () => validExperimentAssignment.retrievedTimestamp );
+		mockedFetchExperimentAssignment.mockImplementationOnce( () =>
+			delayedValue(
+				{
+					ttl: 60,
+					variations: {
+						[ validExperimentAssignment.experimentName ]: validExperimentAssignment.variationName,
+						[ validExperimentAssignment.experimentName +
+						'_repeat' ]: validExperimentAssignment.variationName,
+					},
+				},
+				1
+			)
+		);
+		await expect(
+			Requests.fetchExperimentAssignment( mockedConfig, validExperimentAssignment.experimentName )
+		).rejects.toThrow(
+			'Received multiple experiment assignments while trying to fetch exactly one.'
+		);
+	} );
+
+	it( 'should throw for no experiments in the response', async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( null, 1 ) );
+		spiedMonotonicNow.mockImplementationOnce( () => validExperimentAssignment.retrievedTimestamp );
+		mockedFetchExperimentAssignment.mockImplementationOnce( () =>
+			delayedValue(
+				{
+					ttl: 60,
+					variations: {},
+				},
+				1
+			)
+		);
+		await expect(
+			Requests.fetchExperimentAssignment( mockedConfig, validExperimentAssignment.experimentName )
+		).rejects.toThrow( 'Received no experiment assignments while trying to fetch exactly one.' );
+	} );
+
+	it( 'should throw for response experiment not matching the requested name', async () => {
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( null, 1 ) );
+		spiedMonotonicNow.mockImplementationOnce( () => validExperimentAssignment.retrievedTimestamp );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		await expect(
+			Requests.fetchExperimentAssignment(
+				mockedConfig,
+				validExperimentAssignment.experimentName + '_different_name'
+			)
+		).rejects.toThrow(
+			`Newly fetched ExperimentAssignment's experiment name does not match request.`
+		);
+	} );
+
+	it( 'should throw for returned experiment not being alive', async () => {
+		jest.spyOn( ExperimentAssignments, 'isAlive' ).mockImplementationOnce( () => false );
+		mockedGetAnonId.mockImplementationOnce( () => delayedValue( null, 1 ) );
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		await expect(
+			Requests.fetchExperimentAssignment( mockedConfig, validExperimentAssignment.experimentName )
+		).rejects.toThrow( `Newly fetched experiment isn't alive.` );
+	} );
+} );
+
+describe( 'isFetchExperimentAssignmentResponse', () => {
+	it( 'should return true for valid reponses', () => {
+		expect(
+			Requests.isFetchExperimentAssignmentResponse( {
+				ttl: 0,
```

Comment:
I wonder if this should be a valid response or if we should require `ttl` to be positive...

Reply:
Yeah good point, if the `ttl` is positive that should safe guard us against too many requests being sent.